internal/table/table.go

```diff
@@ -0,0 +1,190 @@
+package table
+
+import (
+	"math"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/sliceutils"
+	"github.com/ActiveState/cli/internal/termutils"
+)
+
+const dash = "\u2500"
+const linebreak = "\n"
+const linebreakRune = '\n'
+const padding = 2
+
+type FormatFunc func(string, ...interface{}) string
+
+type row struct {
+	columns []string
+}
+
+type Table struct {
+	headers []string
+	rows    []row
+}
+
+func New(headers []string) *Table {
+	return &Table{headers, []row{}}
+}
+
+func (t *Table) AddRow(vs ...[]string) *Table {
+	for _, v := range vs {
+		t.rows = append(t.rows, row{v})
+	}
+	return t
+}
+
+func (t *Table) Render() string {
+	if len(t.rows) == 0 {
+		return ""
+	}
+
+	termWidth := termutils.GetWidth()
+	colWidths, total := t.calculateWidth(termWidth)
+
+	out := ""
+	out += "[NOTICE]" + renderRow(t.headers, colWidths) + "[/RESET]" + linebreak
+	out += "[DISABLED]" + strings.Repeat(dash, total) + "[/RESET]" + linebreak
+	for _, row := range t.rows {
+		out += renderRow(row.columns, colWidths) + linebreak
+	}
+
+	return strings.TrimRight(out, linebreak)
+}
+
+func (t *Table) calculateWidth(maxTotalWidth int) ([]int, int) {
+	// Calculate required width of each column
+	minWidth := padding * 2
```

Comment:
We are using this `padding * 2` value in a lot of places. Why not just have the padding be 4?

Reply:
It is chosen that way because the padding happens on both sides of the text (left and right).  At some point we might want to change it that the first and/or last column are handled differently actually (such that they are padded on side only)...